### PR TITLE
Create Multi-Select feature for CountryDistance tab.

### PR DESCRIPTION
**Pull request by vlle1**
* changed some major parts about how data and map are passed around. 
They are now stored in App and passed down to the different components.
This resolves some errors (component rendering without data, all countries gray)

* fixed some weird react errors (no unique keys) and removed some console.logs

* Implemented MultiSelect with Checkboxes
CountryDistance now has another state, which is a subset of the cathegories. The user can alter it by using checkboxes (and an select all/select none button). The distances are now only calculated with regard to the selected cathegories.

### CODE REVIEW

**Comment by vlle1** [comment]
#47 #43 are solved with this PR.

**Review by Willenbrink — CHANGES_REQUESTED** [review]
Looks good! Got some minor things which maybe should be addressed. The timeout thing might be a bit more difficult, it's fine to fix that later I guess.

File: src/features/CountryDistance/CountryDistance.jsx [hunk 1]
```diff
@@ -17,124 +14,197 @@ import { useRef } from "react";
...
-function countryToColor(country, selected) {
-  if (!country.hasData)
-    return colorScheme.noData;
-  if(!selected) {
-    return colorScheme.middle;
-  }
+/*
+ * selectedCathegories: array of ids (strings) of cathegories. country to draw: country-object, selected: country-object
+ */
+const countryToColor = (selectedCathegories) => (country, selected) => {
```
[the remark this answers is not in the document]
Comment (vlle1) [reply src/features/CountryDistance/CountryDistance.jsx:20]: fixed.

File: src/features/CountryDistance/CountryDistance.jsx [hunk 2]
```diff
@@ -17,124 +14,197 @@ import { useRef } from "react";
...
   // Temporary fix for map not rendering on start
-  useEffect(()=>{
+  useEffect(() => {
     async function mount() {
-      await setTimeout(()=>{
-        setSvgHasMounted(activeTab)
-      }, 300)
+      setTimeout(() => {
+        setSvgHasMounted(isActiveTab);
+      }, 300);
```
[the remark this answers is not in the document]
Comment (vlle1) [reply src/features/CountryDistance/CountryDistance.jsx:66]: yes, fiddled around a bit. Maybe we could initialize stuff with the SVG in App.jsx and pass it down, but currently we still need this.

File: src/features/WorldMap/WorldMap.jsx [hunk 3]
```diff
@@ -86,7 +78,7 @@ export default function WorldMap({activeTab}) {
           {svgHasMounted &&
            <>
               <LineDraw
-                data={mapData}
+                data={map}
```
[the remark this answers is not in the document]
Comment (vlle1) [reply src/features/WorldMap/WorldMap.jsx:81]: I actually don't like that we combine the coordinates for the map per country and the data of the country into the same country-object. This new mix makes it unclear if this is now the map or the data. I rename to map_with_data.

File: src/utils/parseMapJSON.jsx [hunk 4]
```diff
@@ -1,49 +1,45 @@
 import React, { useState, useCallback, useEffect } from "react";
 import { json } from "d3";
 import { feature } from "topojson-client";
-import { get_country_data } from "../model/dataHandler";
 import iso from "iso-3166-1";
 
 const mapJSON = "https://unpkg.com/world-atlas@2.0.2/countries-110m.json";
+//gets a dataHandler-data-object
+var data_lazy = null;
```
[the remark this answers is not in the document]
Comment (vlle1) [reply src/utils/parseMapJSON.jsx:8]: purpose was remaining totally functional behavior.
I removed the caching-functionality as this is only executed once anyway.

**Review by marcusgahlin — APPROVED** [review]
Looks good. I think I have a fix for the timeout issue that Sebastian mentioned.

**Pull request merged**